react-basic-practice/src/contexts/loading.jsx

```diff
@@ -0,0 +1,27 @@
+import React, { useContext, useMemo, useState } from 'react';
+
+const LoadingContext = React.createContext();
+
+const useLoading = () => useContext(LoadingContext);
+
+const LoadingProvider = (props) => {
+  const { children } = props;
+  const [loading, setLoading] = useState(false);
+  const showLoading = () => setLoading(true);
```

Comment:
You just need to pass one function instead of both showLoading and hideLoading .